http-server/src/server.rs

```diff
@@ -113,26 +112,31 @@ pub struct Server {
 	/// Local address
 	local_addr: Option<SocketAddr>,
 	/// Registered methods.
-	root: RpcModule,
+	methods: Methods,
 	/// Max request body size.
 	max_request_body_size: u32,
 	/// Access control
 	access_control: AccessControl,
 }
 
 impl Server {
-	/// Register a new RPC method, which responds with a given callback.
-	pub fn register_method<F, R>(&mut self, method_name: &'static str, callback: F) -> Result<(), Error>
-	where
-		R: Serialize,
-		F: Fn(RpcParams) -> Result<R, CallError> + Send + Sync + 'static,
-	{
-		self.root.register_method(method_name, callback)
+	/// Register all [`Methods`] from an [`RpcModule`] on this server. In case a method already is registered with the same name,
+	/// the method is not added. Returns the number of methods registered on the server after the new module's methods
+	/// were added.
+	pub fn register_module<Context>(&mut self, module: RpcModule<Context>) -> usize {
+		for (name, method) in module.into_methods() {
+			if !self.methods.contains_key(name) {
+				self.methods.insert(name, method);
+			} else {
+				log::warn!("Can't add method `{}`: already exists. Skipping", name);
+			}
+		}
+		self.methods.len()
 	}
 
-	/// Register all methods from a module on this server.
-	pub fn register_module(&mut self, module: RpcModule) -> Result<(), Error> {
-		self.root.merge(module)
+	/// Returns a reference to the [`Methods`] registered on the server.
```

Comment:
hmm, when would a user need to access `Methods`?

isn't better just to return `method names` here? but I like the idea

Reply:
I'll remove this actually, good catch, was part of a previous version. We can always add new methods, removing them is harder.

Reply:
…actually no. I am using it tests atm and it I think it's a bit odd that the `Server` offers no introspection at all atm. :/ 
I had this on the `RpcModule` for a while and got confused, sorry for the noise.